packages/core/src/bases/main-wallet.ts

```diff
@@ -0,0 +1,97 @@
+import { ChainName, ChainRegistry, State } from '../types';
+import { MainWalletData } from '../types';
+import { ChainWalletBase } from './chain-wallet';
+import { StateBase } from './state';
+
+export abstract class MainWalletBase<
+  A,
```

Comment:
instead of A, B, C can we use descriptive names, e.g WalletClient, etc, etc

Reply:
ok